### DISCUSSION TITLE
Error message due to 'Blocked' field (Organisation template - read for workaround)

### DISCUSSION THREAD

**Discussion by nbrown02** [nbrown02/FlowViz#58] [Q&A] [answered]
Sometimes, when using the Organisation template, you may get an error that looks like so:

![thumbnail_image001](https://user-images.githubusercontent.com/29369962/158578762-0d29a728-bc17-4091-87a9-2ef5aa763b97.png)

The reason for this is that there are no projects within your collection/organisation that are using the 'Scrum' process template (for example, teams may just be using Agile or Basic and Agile). There is a workaround for this...

**Comment by nbrown02** [comment, marked as answer]
The 'Blocked' field doesn't exist in the Agile or Basic process templates and, unfortunately due to the tables FlowViz queries when getting 'blocked' data, I can't pull all fields in then filter them to account for those process templates that have it missing. 

So to get the query working, 

1. Hit close

<img width="1440" alt="Screenshot 2022-03-16 at 11 59 00" src="https://user-images.githubusercontent.com/29369962/158586641-a5716a31-2364-47ef-b622-f461b86eaeb0.png">

2. Select Transform Data

<img width="1440" alt="Screenshot 2022-03-16 at 12 03 51" src="https://user-images.githubusercontent.com/29369962/158586410-620ffe2e-4e11-4901-880b-7e8ccb811d82.png">

4. Delete the text highlighted:

![Unknown-5](https://user-images.githubusercontent.com/29369962/158579654-8c8de204-a7a1-417a-927c-3fe2cd6d412c.png)

5. Make sure to do this for the following tables - _WorkItems Bl0cked, WorkItems Daily Blocked, WorkItems In Progress, WorkItems Flow Efficiency_

6. Once all changes are made, hit 'Close and apply' then your charts will load :)

**Reply by neensKB** [reply]
Hi Nick! I followed these instructions and the tables loaded inside Power Query Editor but when applying them, I get different errors (screengrabs attached)
![FlowViz error 1](https://github.com/nbrown02/FlowViz/assets/107828094/3ef0307d-5ac5-46bc-93d9-3761957fd921)
![FlowViz error 2](https://github.com/nbrown02/FlowViz/assets/107828094/f9f2134c-52e0-4e22-a070-f1edc527028b)

**Reply by nbrown02** [reply]
Hi there,

What process template is in use across all projects? This might happen if no projects are using a process template that contains the blocked field...

**Reply by neensKB** [reply]
All projects use a variation of the Agile template

**Reply by nbrown02** [reply]
Hmm ok - my only suggestion would be to go into Power Query and removed "Blocked" from all queries. Are you familiar with Power Query or need some screenshots to help guide you?

**Reply by neensKB** [reply]
I did that already as per the original workaround you shared - I’m getting the errors after removing it. The tables seem to load OK inside Power Query editor and I get the spinny circles for a minute or so after clicking “close & apply” but then they fail with the above errors.

It’s all good if there isn’t much else to try out - it actually helps me make my case for our need to reconfigure our ADO instance 🤓 (it had been neglected/left to grow organically for a few years before we started paying attention to it again…)

**Reply by nbrown02** [reply]
Ok so with _WorkItems In Progress_ table, the last applied step needs Blocked removed from it:

![image](https://github.com/nbrown02/FlowViz/assets/29369962/d6915087-f9d6-4d44-a9e4-ff72aa08122f)

For _WorkItems FlowEfficiency_ Blocked should be removed from the query then it will load
